index.html

```diff
@@ -417,42 +384,57 @@
     };
 
     callback ThingEventListener = void (ConsumedThing thing);
+
+    enum ConsumedThingEvents {
+        "propertychanged",
+        "propertyremoved",
+        "propertyadded",
+        "actionchanged"
+    };
   </pre>
   </section>
 
   <section> <h2>The Thing Server API</h2>
   <pre class="idl">
     interface ExposedThing {
+
+      // define TD modifiers
       ExposedThing addProperty(ThingProperty property);
+      ExposedThing removeProperty(ThingProperty property);
+
       ExposedThing addAction(ThingAction action);
+      ExposedThing removeAction(ThingAction action);
+
       ExposedThing addEvent(ThingEvent event);
+      ExposedThing removeEvent(ThingEvent event);
 
-      Promise&lt;void&gt; register();
+      Promise&lt;void&gt; register(optional USVString directory);
       Promise&lt;void&gt; unregister();
 
       Promise&lt;void&gt; start();
       Promise&lt;void&gt; stop();
 
       Promise&lt;void&gt; emitEvent(DOMString eventName, any payload);
 
-      // define request handlers
-      ExposedThing onPropertyRetrieve(PropertyRetrieveHandler handler);
-      ExposedThing onPropertyUpdate(PropertyHandler handler);
+      // define request handlers (one per request type, so no events here)
+      ExposedThing onRetrieveProperty(PropertyHandler handler);
+      ExposedThing onAddProperty(PropertyHandler handler);
+      ExposedThing onRemoveProperty(PropertyHandler handler);
```

Comment:
Actually we don't need a separate `PropertyUpdateHandler` since in the request the `ThingProperty` parameter does have a requested new `value`, and the old value is what it is now, that is not changed. This is a request handler, not an event for notifications.